File: Chapter08/C#/Project/WebServices/2-ASPNETWebService/ASMXWebService/SilverlightResource.asmx.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using System.Web.Services;
using System.Data.SqlClient;

namespace ASMXService
{
    /// <summary>
    /// Summary description for Resource Service
    /// </summary>
    [WebService(Namespace = "http://tempuri.org/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [System.ComponentModel.ToolboxItem(false)]
    public class SilverlightResource : WebService
    {
        private ResourceInfo Resource;
        public SilverlightResource()
        {
            Resource.Author = "";
            Resource.Title = "";
            Resource.Image = "";
            Resource.URL = "";
            Resource.Type = "";
        }
        private void GetResourceData(int ResourceId)
        {
            // This is where you use your business components. 
            // Method calls on Business components are used to populate the data.
            
            Resource.Title = "Silverlight 3 How To";
            Resource.Author = "Rajesh Lal";
            Resource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/S3HT.jpg";
            Resource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628";

            try
            {
                string query = "SELECT * FROM Resource where ID=" + ResourceId;

                clsDataAccess mydataAccess = new clsDataAccess();
                mydataAccess.openConnection();
                SqlDataReader mydr = mydataAccess.getData(query);
                if (mydr.HasRows)
                {
                    while (mydr.Read())
                    {
                        Resource.Title = mydr.GetValue(1).ToString();
                        Resource.Author = mydr.GetValue(2).ToString();
                        Resource.Image = mydr.GetValue(3).ToString();
                        Resource.URL = mydr.GetValue(4).ToString();
                        if (Convert.ToInt32(mydr.GetValue(5))==1) Resource.Type = "Book";
                        if (Convert.ToInt32(mydr.GetValue(5)) == 2) Resource.Type = "Blog";
                        if (Convert.ToInt32(mydr.GetValue(5)) == 3) Resource.Type = "Tutorial";
                        if (Convert.ToInt32(mydr.GetValue(5)) == 4) Resource.Type = "Video";
                        
                    }
                }
                else
                {
                    Resource.Title = "Error: No data for that ID" ;
                    Resource.Author = "Try other ID's between 1 and 16";
                    Resource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg";
                    Resource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628";
                }

                mydr.Close();
                mydataAccess.closeConnection();
            }
            catch (Exception e)
            {
                Resource.Title = "Error: " + e.Message;
                Resource.Author = "Occured while connecting to the Database ";
                Resource.Image = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg";
                Resource.URL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628";
            }
        }
        [WebMethod(Description = "This method call will get the Resource information from the database for a given Resource ID.", EnableSession = false)]
        public ResourceInfo GetResource(int id)
        {
            GetResourceData(id);
            ResourceInfo localResource = new ResourceInfo();
            localResource.Author = Resource.Author;
            localResource.Title= Resource.Title;
            localResource.Image = Resource.Image;
            localResource.URL= Resource.URL;
            localResource.Type= Resource.Type;
            return localResource;
        }
    }
    public struct ResourceInfo
    {
        public string Title;
        public string Author;
        public string Image;
        public string URL;
        public string Type; 
    }

}

File: Chapter08/VB/VBProject_08/WebServices/2-ASPNETWebService/ASMXWebService/SilverlightResource.asmx.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Web
Imports System.Web.Services
Imports System.Data.SqlClient

'Namespace ASMXService

''' <summary>
''' Summary description for Resource Service
''' </summary>
<WebService(Namespace:="http://tempuri.org/")>
<WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)>
<System.ComponentModel.ToolboxItem(False)>
Public Class SilverlightResource
    Inherits WebService

    Private oResource As ResourceInfo

    Public Sub New()

        oResource.Author = ""
        oResource.Title = ""
        oResource.sImage = ""
        oResource.sURL = ""
        oResource.sType = ""
    End Sub '   New

    Private Sub GetResourceData(ResourceId As Integer)

        '  This is where you use your business components.
        '  Method calls on Business components are used to populate the data.

        oResource.Title = "Silverlight 3 How To"
        oResource.Author = "Rajesh Lal"
        oResource.sImage = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/S3HT.jpg"
        oResource.sURL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628"

        Try

            Dim query As String = "SELECT * FROM Resource where ID=" + CStr(ResourceId)
            Dim mydataAccess As clsDataAccess = New clsDataAccess()

            mydataAccess.openConnection()

            Dim mydr As SqlDataReader = mydataAccess.getData(query)

            If (mydr.HasRows) Then
                While (mydr.Read())
                    oResource.Title = mydr.GetValue(1).ToString()
                    oResource.Author = mydr.GetValue(2).ToString()
                    oResource.sImage = mydr.GetValue(3).ToString()
                    oResource.sURL = mydr.GetValue(4).ToString()

                    Select Case (Convert.ToInt32(mydr.GetValue(5)))
                        Case 1
                            oResource.sType = "Book"
                        Case 2
                            oResource.sType = "Blog"
                        Case 3
                            oResource.sType = "Tutorial"
                        Case 4
                            oResource.sType = "Video"
                    End Select
                End While   '
            Else
                oResource.Title = "Error: No data for that ID"
                oResource.Author = "Try other ID's between 1 and 16"
                oResource.sImage = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg"
                oResource.sURL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628"
            End If

            mydr.Close()
            mydataAccess.closeConnection()
        Catch e As Exception
            oResource.Title = "Error: " + e.Message
            oResource.Author = "Occured while connecting to the Database "
            oResource.sImage = "http://silverlightfun.com/Samples/Chapter-07/AdoData/images/error.jpg"
            oResource.sURL = "http://www.amazon.com/Silverlight-3-How-Rajesh-Lal/dp/0672330628"
        End Try
    End Sub '   GetResourceData

    <WebMethod(Description:="This method call will get the Resource information from the database for a given Resource ID.", EnableSession:=False)>
    Public Function GetResource(id As Integer) As ResourceInfo

        GetResourceData(id)

        Dim localResource As ResourceInfo = New ResourceInfo()

        localResource.Author = oResource.Author
        localResource.Title = oResource.Title
        localResource.sImage = oResource.sImage
        localResource.sURL = oResource.sURL
        localResource.sType = oResource.sType
        Return localResource
    End Function  '   GetResource
End Class   '   SilverlightResource

Public Structure ResourceInfo

    Public Title As String
    Public Author As String
    Public sImage As String
    Public sURL As String
    Public sType As String
End Structure
'End Namespace
' ..\Project_08\WebServices\2-ASPNETWebService\ASMXWebService\SilverlightResource.asmx.cs
